CodeStarterCamp_Week3/Step1.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  Step1.swift
+//  CodeStarterCamp_Week3
+//
+//  Created by 홍성준 on 2023/08/02.
+//
+
+import Foundation
+
+class Person {
+    var name: String
+    var asset: Int = 0
+    init(name: String, asset: Int) {
+        self.name = name
+        self.asset = asset
+    }
+    
+    func buyCoffee(coffeeShop: CoffeeShop, menus: [Coffee]) {
+        coffeeShop.getOrder(menus: menus)
+        for menu in menus {
+            if let price = coffeeShop.menus[menu.rawValue] {
+                self.asset -= price
+            }
+        }
+    }
+    
+}
+
+class CoffeeShop {
+    var barista: Person?
+    var sales: Int = 0
+    var menus: Dictionary<String, Int> = [:]
+    var pickUpTable: Array<String> = []
+    init() {
+        self.menus = [Coffee.americano.rawValue: 4000,
+                      Coffee.latte.rawValue: 4500,
+                      Coffee.iceTea.rawValue: 5000]
+    }
+    func getOrder(menus: [Coffee]) {
+        for menu in menus {
+            self.makeCoffee(menu.rawValue)
+        }
+    }
+    
+    func makeCoffee(_ menu: String) {
+        self.pickUpTable.append(menu)
+        if let sale = menus[menu] {
+            self.sales += sale
+        }
+    }
+}
+
+enum Coffee: String {
+    case americano = "americano"
+    case latte = "latte"
+    case iceTea = "iceTea"
+}
```

Comment:
열거형의 원시값을 활용해주셨네요 👏
원시값을 직접 할당하지 않고 String 타입이라는 것만을 알려준다면 원시값이 어떻게 작동하는지 확인해볼까요?
```suggestion
enum Coffee: String {
    case americano 
    case latte 
    case iceTea
}
```

Reply:
타입만 알려준다면 문자열인 경우 case 이름과 동일한 원시값이 자동으로 만들어집니다. 제가 쓴 코드의 경우에는 굳이 문자열을 따로 쓰지 않아도 되겠네요!

Reply:
참고로 원시값의 타입이 Int인 경우에 원시값을 할당하지 않는다면
제일 위쪽 case의 원시값이 0부터 시작해서 1씩 증가하게 됩니다~!